include/battle.h

```diff
@@ -366,11 +366,12 @@ struct AiLogicData
     u8 effectiveness[MAX_BATTLERS_COUNT][MAX_BATTLERS_COUNT][MAX_MON_MOVES]; // attacker, target, moveIndex
     u8 moveAccuracy[MAX_BATTLERS_COUNT][MAX_BATTLERS_COUNT][MAX_MON_MOVES]; // attacker, target, moveIndex
     u8 moveLimitations[MAX_BATTLERS_COUNT];
-    bool8 shouldSwitchMon; // Because all available moves have no/little effect. Each bit per battler.
+    bool8 shouldSwitchIfBadMoves; // Because all available moves have no/little effect. Each bit per battler.
     u8 monToSwitchId[MAX_BATTLERS_COUNT]; // ID of the mon to switch.
     bool8 weatherHasEffect; // The same as WEATHER_HAS_EFFECT. Stored here, so it's called only once.
     u8 mostSuitableMonId[MAX_BATTLERS_COUNT]; // Stores result of GetMostSuitableMonToSwitchInto, which decides which generic mon the AI would switch into if they decide to switch. This can be overruled by specific mons found in ShouldSwitch; the final resulting mon is stored in AI_monToSwitchIntoId.
     struct SwitchinCandidate switchinCandidate; // Struct used for deciding which mon to switch to in battle_ai_switch_items.c
+    u8 shouldSwitch[MAX_BATTLERS_COUNT]; // Stores result of ShouldSwitch, which decides whether a mon should be switched out
```

Comment:
This could be a bitfield as well, right?

Reply:
I am almost entirely unfamiliar with implementing new bitfields, so I don't know :D

I have just copied existing patterns as needed up to this point whenever they come up

Reply:
I haven't checked but if the result of shouldSwitch is either true or false then you could use it here. For reference it is used for `shouldSwitchIfBadMoves`

Reply:
ShouldSwitch is always either true or false, sometimes it emits a value to other functions but never as a `return` and also never for this new field.

I definitely see what you mean with the pattern, so if I'm not mistaken it'd be 

```diff
- AI_DATA->shouldSwitch[battler] = ShouldSwitch(battler, FALSE);
+ AI_DATA->shouldSwitch |= ShouldSwitch(battler, FALSE) << battler
```
to initialize it, and then to reference the result later it'd be

```diff
- AI_DATA->shouldSwitch[battler]
+ AI_DATA->shouldSwitch & (1u << battler)
```
is that correct?

Is this significantly better for space? I find it much more opaque to read lol

Reply:
it needs only 8bits compared to 32, also it might be faster? And yeah if an optimization like that is possible I would do it because it doesn't cost much. 
You could do 
```
if (ShouldSwitch(battler, FALSE))
    AI_DATA->shouldSwitch |= (1u << battler)

Reply:
Pushed, let me know if anything looks wrong but it seems to work :)